engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -418,20 +425,18 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
 
   private def validateCwlNamespace(source: WorkflowSourceFilesCollection,
                                    workflowOptions: WorkflowOptions,
-                                   pathBuilders: List[PathBuilder]): ErrorOr[ValidatedWomNamespace] = {
+                                   pathBuilders: List[PathBuilder]): Parse[ValidatedWomNamespace] = {
     // TODO WOM: CwlDecoder takes a file so write it to disk for now
     import better.files._
-    import cats.syntax.either._
 
     val cwlFile = File.newTemporaryFile(prefix = workflowIdForLogging.toString).write(source.workflowSource)
 
     try {
       for {
-        wf <- CwlDecoder.decodeAllCwl(cwlFile).map {
-          _.select[Workflow].get
-        }.value.unsafeRunSync.toValidated
-        executable <- wf.womExecutable
-        graph <- executable.graph
+        cwl <- CwlDecoder.decodeAllCwl(cwlFile)
+        wf <- EitherT{ IO{ cwl.select[Workflow].toRight(NonEmptyList.one(s"expected a workflow but got a $cwl")) }}
+        executable <- EitherT{ IO{ wf.womExecutable }}
+        graph <- EitherT{ IO{ executable.graph.toEither }}
```

Comment:
Is it possible to sugar the `EitherT { IO { ... }}`? There's a lot of that in this PR.

Reply:
fixed it up using [`fromEither[IO]`](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/EitherT.scala#L351)